src/Features/Core/Portable/SemanticSearch/AbstractSemanticSearchService.cs

```diff
@@ -0,0 +1,436 @@
+        {
+            var trace = new StackTrace(e, fNeedFileInfo: true);
+            var frames = trace.GetFrames();
+            var displayFrames = frames;
+            var skippedFrameCount = 0;
+
+            try
+            {
+                var hostAssembly = typeof(AbstractSemanticSearchService).Assembly;
+                var displayFramesEnd = frames.Length;
+                var foundPosition = false;
+                for (var i = 0; i < frames.Length; i++)
+                {
+                    var frame = frames[i];
+
+                    if (frame.GetMethod() is { } method)
+                    {
+                        var frameAssembly = method.DeclaringType?.Assembly;
+                        if (frameAssembly == hostAssembly)
+                        {
+                            displayFramesEnd = i;
+                            break;
+                        }
+
+                        if (!foundPosition &&
+                            frameAssembly == queryAssembly &&
+                            frame.GetFileName() is { } fileName &&
+                            frame.GetFileLineNumber() is > 0 and var line &&
+                            frame.GetFileColumnNumber() is > 0 and var column)
+                        {
+                            position = new LinePosition(line - 1, column - 1);
+                            foundPosition = true;
+                        }
+                    }
+                }
+
+                // display last StackDisplayDepthLimit frames preceding the host frame:
+                skippedFrameCount = Math.Max(0, displayFramesEnd - StackDisplayDepthLimit);
+                displayFrames = frames[skippedFrameCount..displayFramesEnd];
+            }
+            catch
+            {
+                // nop
+            }
+
+            formattedTrace =
+            [
+                new TaggedText(tag: TextTags.Text, (skippedFrameCount > 0 ? "   ..." + Environment.NewLine : "") + GetStackTraceText(displayFrames))
+            ];
+        }
+        catch
+        {
+            formattedTrace = [];
+        }
+    }
+
+    private static string GetStackTraceText(IEnumerable<StackFrame> frames)
+    {
+#if NET8_0_OR_GREATER
+        return new StackTrace(frames).ToString();
+#else
+        var builder = new StringBuilder();
+        foreach (var frame in frames)
+        {
+            builder.Append(new StackTrace(frame).ToString());
+        }
+
+        return builder.ToString();
+#endif
+    }
+
+    private static bool TryGetFindMethod(Assembly queryAssembly, [NotNullWhen(true)] out MethodInfo? method, out string? error, out string[]? errorMessageArgs)
+    {
+        // TODO: Use Compilation APIs to find the method
+
+        method = null;
+        error = null;
+        errorMessageArgs = null;
+
+        Type? program;
+        try
+        {
+            program = queryAssembly.GetType(WellKnownMemberNames.TopLevelStatementsEntryPointTypeName, throwOnError: false);
```

Comment:
Consider only support top level method for simplicity. We don't seems to have a way to let user specify with of multiple suitable methods they want to run anyway.

Reply:
I'll follow up on this.